[b1inmeister] iP

b1inmeister opened the pull request.

b1inmeister committed:
Add Level 0 increment.

Chatbot is named Duke, with an accompanying introduction message.

Level 0 increment for iP requires renaming the chatbot and adding greeting and exit messages.

Let's rename the chatbot to Amadinho, and add greeting and exit messages to Amadinho.

b1inmeister committed:
Update code to echo inputs

Import Scanner class for inputs
Create function echo() that takes input then prints it
Modify intro and exit messages to functions intro() and exit()
Put all functions into main()

b1inmeister committed:
Update code with list functionality

Create a new array of Strings called toDo
Modify echo() to conduct specific commands
Add function add() to put instructions into toDo
Add function list() to print all contents in toDo

b1inmeister committed:
Add new class

Create new class Task for the tasks in toDo

b1inmeister committed:
Update Task class

Add description getter function
Add markAsDone() function to update isDone when task is complete

b1inmeister committed:
Update main code

Modify add() and list() to use the Task class for the list
Update main() to use the Task class in the toDo array

b1inmeister committed:
Update main code to mark Task as done

Create mark() to mark a Task in the toDo array as complete
Add unmark() template for future implementation
Modify readInput() to accept mark and unmark commands

b1inmeister committed:
Update task class with new functionality

Create markAsUndone() to revert status to not complete yet

b1inmeister committed:
Update main code to unmark Task

Add functionality for unmark() to revert Task to not complete yet

b1inmeister committed:
Update mark() and unmark() with exceptions

Add exceptions to do while loops when index is beyond Tasks in list

b1inmeister committed:
Update main code to coding standards

Remove intro() and exit() for simplicity
Add functionality of intro() and exit() directly into main()
Check and modify code to follow coding standards
Remove redundant README file

b1inmeister committed:
Add new inherited class for different Tasks

Implement Todo class for Tasks with no date or time
Implement Deadline class for Tasks that have deadlines
Implement Event class for Tasks that have start and end time

b1inmeister committed:
Modify String output in Task class

Override toString() in Task class
Change list() in main code to use new toString()

b1inmeister committed:
Edit toString() in Task class

Remove one space from returned String in toString()

b1inmeister committed:
Edit main code to use new Tasks

Modify add() to use new inherited classes
Add output messages for add()
Modify output message for mark() and unmark()
Modify output message for list()
Remove space from list printing in list()

b1inmeister committed:
Edit auto IO test files

Change name in runtest files from Duke to Amadinho

b1inmeister committed:
Edit main code for refactoring Round 1

Move messages and errors to separate methods
Convert magic strings and numbers to constants
Modify execution of commands executeCommand()
Modify printing list of tasks commandList()
Modify addition of Todo tasks commandTodo()
Add insertion of task into list as method

xyanjun02 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,232 @@
+    public static final String MESSAGE_ERROR_INVALID_COMMAND = "Invalid command. Please try again.";
+    public static final String MESSAGE_ERROR_OUTOFBOUNDS = "Number provided is not in the list.";
+
+    public static final String BORDER_LINE = "____________________________________________________________";
```
I like the well-structured and organized approach in defining the constants

b1inmeister replied:
Thank you!

b1inmeister committed:
Edit main code for refactoring Round 2

Convert input reading statements to methods
Create boolean to detect when to end Amadinho
Refactor printing list portion to another method
Rework marking and unmarking methods
Refactor mark messages to another method

xyanjun02 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,232 @@
+
+    public static void executeCommand(Task[] taskList, String userCommand, String information) {
+        switch (userCommand) {
+        case COMMAND_LIST:
```
The indentation for switch statement is correct. Nice!

b1inmeister replied:
Thank you!

xyanjun02 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,232 @@
+        for (int arrayCounter = COUNTER_START; arrayCounter < taskList.length; arrayCounter++) {
+            if (taskList[arrayCounter] == null) {
+                break;
+            } else {
```
Correct if - else statement indentation, nice!

b1inmeister replied:
Thank you!

xyanjun02 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,232 @@
+        Scanner in = new Scanner(System.in);
+        int taskPosition = in.nextInt();
+
+        // counter for do while loop
```
Comment indented relative to it's position in the code. Nice!

xyanjun02 reviewed (commented):
Looks good in general!

randust commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+        }
+
+        exitMessage();
+    }
```
I like your structure and the code is well organized to become very easy to read at the high level

randust commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+        return userCommand.equals(COMMAND_BYE);
+    }
+
+    public static void executeCommand(Task[] taskList, String userCommand, String information) {
+        switch (userCommand) {
+        case COMMAND_LIST:
+            commandList(taskList);
+            break;
+        case COMMAND_MARK:
+            commandMark(taskList, information);
+            break;
+        case COMMAND_UNMARK:
+            commandUnmark(taskList, information);
+            break;
+        case COMMAND_TODO:
+            commandTodo(taskList, information);
+            break;
+        case COMMAND_DEADLINE:
+            commandDeadline(taskList, information);
+            break;
+        case COMMAND_EVENT:
+            commandEvent(taskList, information);
+            break;
+        default:
+            errorInvalidCommand();
+            break;
```
the case switch statement is highly readable as well

b1inmeister replied:
Thank you!

randust commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+    public static void printList(Task[] taskList) {
+        int taskCounter = LIST_COUNTER_START;
+
+        for (int arrayCounter = COUNTER_START; arrayCounter < taskList.length; arrayCounter++) {
+            if (taskList[arrayCounter] == null) {
+                break;
+            } else {
+                System.out.println(taskCounter + "." + taskList[arrayCounter]);
+                taskCounter++;
+            }
+        }
```
the naming of arrayCounter and taskCounter is a bit vague. Do you mean taskArrayCounter?

b1inmeister replied:
Yes, that is what I meant. I have changed the for loop to an enhanced for loop which removes the need for taskArrayCounter. Thank you for the input!

randust commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+        System.out.println(BORDER_LINE);
+
+        if (toMark) {
+            System.out.println(MESSAGE_MARK_COMPLETE);
+        } else {
+            System.out.println(MESSAGE_UNMARK_COMPLETE);
+        }
+
+        System.out.println(taskCount + "." + taskList);
+        System.out.println(BORDER_LINE);
+    }
+
+    public static void addCommandMessage(Task newTask, int arrayCounter) {
+        int totalTasks = arrayCounter + ARRAY_INCREMENT;
+
+        System.out.println(BORDER_LINE);
+        System.out.println(MESSAGE_ADDED_TASK);
+        System.out.println("  " + newTask);
+        System.out.println(printTotalTasks(totalTasks));
+        System.out.println(BORDER_LINE);
+    }
+
+    public static String printTotalTasks(int totalTasks) {
+        return "Now you have " + totalTasks + " tasks in the list.";
+    }
+
+
+    /**
+     * ERROR-RELATED METHODS
+     */
+
+    public static void errorOutOfBounds() {
+        throw new IndexOutOfBoundsException(MESSAGE_ERROR_OUTOFBOUNDS);
+    }
+
+    public static void errorInvalidCommand() {
+        System.out.println(BORDER_LINE);
+        System.out.println(MESSAGE_ERROR_INVALID_COMMAND);
+        System.out.println(BORDER_LINE);
+    }
```
the printing and error handling is well organized

CRL006 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+    public static void printList(Task[] taskList) {
+        int taskCounter = LIST_COUNTER_START;
+
+        for (int arrayCounter = COUNTER_START; arrayCounter < taskList.length; arrayCounter++) {
```
You avoided deep nesting in your code! Good job!

b1inmeister replied:
Thank you!

CRL006 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+        }
+    }
+
+    public static void commandList(Task[] taskList) {
```
You avoided long methods! Good job!

b1inmeister replied:
Thank you!

CRL006 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+    public static final int ARRAY_INCREMENT = 1;
+    public static final int COUNTER_START = 0;
+
+    public static final String COMMAND_BYE = "bye";
```
You structured the code logically, splitting the strings for commands and messages! Good job!

b1inmeister replied:
Thank you!

CRL006 commented on src/main/java/Amadinho.java:
```diff
@@ -0,0 +1,235 @@
+    }
+    
+    // incomplete
+    public static void commandDeadline(Task[] taskList, String information) {
```
You avoided having multiple levels of code abstraction within a code fragment! Good job!

b1inmeister committed:
Edit main code for refactoring Round 3

Add constants for command identifiers and their length
Modify printing list loop to enhanced for loop
Rework deadline and event methods
Refactor string manipulation to new methods
Edit welcome message to simpler version

b1inmeister committed:
Edit main code for slight error handling

Add custom messages for invalid commands
Add switch for custom messages
Add error handling for list and mark
Customise notification messages

b1inmeister committed:
Add exceptions for error handling

Create exception for incorrect command syntax
Create exception for when trying to print empty list

b1inmeister committed:
Edit main code for proper error handling

Add new error methods to throw exceptions
Modify error messages
Add try catch statements for command execution methods

b1inmeister committed:
Merge branch 'branch-Level-5'

b1inmeister committed:
Put classes into separate packages

Group all Task classes into tasktypes package
Group all Exception classes into exception package
Add main class into ui package
Modify auto IO test to read new location of main

b1inmeister committed:
Merge branch 'branch-A-Packages'

b1inmeister committed:
Implement new Constants class

Create new class for constants
Put new class into ui package
Edit main code to import new class

b1inmeister committed:
Edit code for housekeeping

Remove redundant method in Task class
Modify access modifiers in Constants class
Modfiy access modifiers in main class

b1inmeister committed:
modify code for ArrayList

change taskList from array to ArrayList
edit command methods to use ArrayList class methods
remove redundant constants and error method

b1inmeister committed:
Add delete functionality

Add command for deleting Tasks
Add new messages for deleting process
Add error checking for invalid delete format

b1inmeister committed:
Add writing to file functionality

Import required libraries
Create method and method calls for writing to text file
Add method for error printing
Add constant error messages for writing to file
Add constant syntax for text file formatting
Modify Task subclasses with getter methods
Add templates for other file methods

b1inmeister committed:
Add reading from file functionality

Create super method to read from text file
Create sub methods to insert different tasktypes to list
Create method to create new file or directory if needed
Define boolean to distinguish between command input and reading from text
Modify Task class with new getter and setters
Modify old writing to file method to use new getter
Add new constant error message for reading errors

b1inmeister committed:
Modify file IO for marking Tasks

Modify both reading and writing methods to accomodate marking of Tasks
Change symbol of uncompleted Tasks to  _
Create new Task getter for task type

b1inmeister committed:
Merge branch 'branch-Level-6'

b1inmeister committed:
Resolve merge conflicts for Level-7

Modify file IO methods to use ArrayList
Edit deletion method to write to file

b1inmeister committed:
Add error handling for mark and delete

Modify mark and delete methods for index out of bounds
Add proper error message and print method

b1inmeister committed:
Clean code to prepare for JAR

Replace existing integers and strings with constants
Refactor file IO methods

b1inmeister committed:
Modify files for IO testing

Fill in text UI files with input and expected output

b1inmeister committed:
Convert code to follow OOP principles

Consolidate methods into new classes
Allocate new classes to new packages
Edit expected output for OOP IO testing
Modify runtest files for OOP IO testing

b1inmeister committed:
Add date & time recognition

Edit deadline and event methods to recognise date and time
Set inputs of /by, /from and /to to date and time only
Modify deadline and event methods for incorrect input checking
Edit IO testing files for testing of edited methods
Move relevant constants to new constant classes for clarity
Put new constant classes in respective packages

b1inmeister committed:
Add find functionality

Create new find method to check description
Create new exception for when word to find is missing
Refactor list printing method for reusability
Edit IO testing files to test new find method

b1inmeister committed:
Implement in-code documentation

Add JavaDoc comments for all classes
Add JavaDoc comments for all methods

b1inmeister committed:
Merge pull request #1 from b1inmeister/branch-Level-8

Merge branch-Level-8

b1inmeister committed:
Merge branch 'master' into branch-Level-9

b1inmeister committed:
Modify code post-merge

Remove redundant methods
Move constants to new constants class

b1inmeister committed:
Merge branch 'master' into branch-A-JavaDoc

b1inmeister committed:
Merge pull request #2 from b1inmeister/branch-Level-9

Merge branch-Level-9

b1inmeister committed:
Merge branch 'master' into branch-A-JavaDoc

b1inmeister committed:
Merge pull request #3 from b1inmeister/branch-A-JavaDoc

Merge branch-A-JavaDoc

b1inmeister committed:
Edit README.md

Add user guide for Amadinho

b1inmeister committed:
Check code before release

Edit code for housekeeping and organising purposes
Change IO tests for smoke testing

aditichadha1310 reviewed (commented):
very good work!! Keep it up.
